qemu/tests/in_place_upgrade_legacy.py

```diff
@@ -0,0 +1,126 @@
+
+
+class IpuLegacyTest(IpuTest):
+    """
+    Provide basic functions for in place upgrade test cases
+
+    """
+    def __init__(self, test, params, env):
+
+        super(IpuTest, self).__init__()
+        self.session = None
+        self.test = test
+        self.params = params
+        self.env = env
+
+    def pre_upgrade_whitelist(self, test):
+        """
+        Fix known issues after executing pre-upgrade
+
+        """
+        try:
+            # Possible problems with remote login using root account
+            self.session.cmd(self.params.get("fix_permit"))
+            # Answer file missing will be fixed
+            fix_answer = self.params.get("fix_answer_file")
+            self.session.cmd(fix_answer, timeout=1200)
+            fix_answer_sec = self.params.get("fix_answer_section")
+            self.session.cmd(fix_answer_sec, timeout=1200)
+            erase_old_kernel = self.params.get("clean_up_old_kernel")
+            s, output = self.session.cmd_status_output(erase_old_kernel,
+                                                       timeout=1200)
+            error_info = self.params.get("error_info")
+            if re.search(error_info, output):
+                pass
+        except Exception as info:
+            test.fail("Failed to fix known issues in advance")
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Run in place upgrade cases rhel7.9 guest:
+    a) without RHSM
+    1.configure vm
+    2.install leapp tool
+    3.download new rhel content repo
+    4.pre_upgrade test in the vm
+    5.upgrade test in the vm
+    6.check if it's target system
+
+    b) with rhsm
+    1.configure vm
+    2.install leapp tool
+    3.subscribe vm
+    4.pre_upgrade test in the vm
+    5.upgrade test in the vm
+    6.check if it's target system
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    upgrade_test = IpuLegacyTest(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    login_timeout = int(params.get("login_timeout", 360))
+    upgrade_test.session = vm.wait_for_login(timeout=login_timeout)
+    try:
+        # set post_release
+        pre_release = params.get("pre_release")
+        release_chk = params.get("release_check")
+        if pre_release not in upgrade_test.run_guest_cmd(release_chk):
+            test.cancel("your image is not for rhel 7.9 product, please check")
+        post_release = params.get("post_release")
+        if params.get("rhsm_type") == "no_rhsm":
+            # internal repo for doing yum update
+            old_custom_repo = params.get("old_custom_internal_repo_7")
+            upgrade_test.yum_update_no_rhsm(test, old_custom_repo)
+        elif params.get("rhsm_type") == "rhsm":
+            # doing rhsm and update the old system
+            upgrade_test.run_guest_cmd(params.get("prepare_env"))
```

Comment:
Seems `prepare_env ` is not defined.

Reply:
It's in our internal config file.

Reply:
Will give comments for this and make these parameters clearly.